Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -3894,22 +3894,39 @@ static NSStringCompareOptions compareOptionsForFindOptions(WebCore::FindOptions
 
 #if PLATFORM(IOS_FAMILY)
 
-std::optional<FloatRect> UnifiedPDFPlugin::highlightRectForTapAtPoint(FloatPoint pointInRootView) const
+std::pair<URL, FloatRect> UnifiedPDFPlugin::linkURLAndBoundsAtPoint(FloatPoint pointInRootView) const
 {
     RetainPtr annotation = annotationForRootViewPoint(roundedIntPoint(pointInRootView));
     if (!annotation)
-        return std::nullopt;
+        return { };
 
     if (!annotationIsLinkWithDestination(annotation.get()))
-        return std::nullopt;
+        return { };
+
+    return { [annotation URL], pageToRootView([annotation bounds], [annotation page]) };
```

Comment:
It's possible that the annotation is an in-document link, in which case `[annotation URL]` is `nil`.

Reply:
Hmm, it looks like `_tryToCopyLinkURLFromPlugin` does the right thing (bails when the URL is nil)

Reply:
But we probably want to change how we're calling this in `highlightRectForTapAtPoint()`, because we do a no-op for document destination links.

Reply:
Ah, good point — we probably want to continue showing a tap highlight for in-document links then, but not a full context menu. I'll ensure that by changing `highlightRectForTapAtPoint` to:

```
std::optional<FloatRect> UnifiedPDFPlugin::highlightRectForTapAtPoint(FloatPoint pointInRootView) const
{
    // FIXME: We only support tapping on links at the moment. In the future, we might want to
    // support more types of annotations.
    auto [url, rect] = linkURLAndBoundsAtPoint(pointInRootView);
    if (rect.isEmpty())
        return std::nullopt;

    return rect;
}
```